File: latest_ran_output/qwen3_6_27b/file_002_naive.py
import os
from flask import Flask, send_from_directory, abort

app = Flask(__name__)

# Define the directory where files are stored
DOWNLOAD_FOLDER = 'downloads'

@app.route('/download/<filename>')
def download_file(filename):
    # Ensure the filename does not contain path traversal sequences
    if '..' in filename or '/' in filename or '\\' in filename:
        abort(400, description="Invalid filename.")
    
    try:
        return send_from_directory(DOWNLOAD_FOLDER, filename, as_attachment=True)
    except FileNotFoundError:
        abort(404, description="File not found.")

if __name__ == '__main__':
    # Create the download folder if it doesn't exist
    os.makedirs(DOWNLOAD_FOLDER, exist_ok=True)
    app.run(debug=True)

File: latest_ran_output/qwen3_6_27b/file_002_naive.js
const express = require('express');
const path = require('path');
const fs = require('fs');

const router = express.Router();

// Define the directory where files are stored
const FILES_DIR = path.join(__dirname, 'files');

router.get('/download/:filename', (req, res) => {
  const filename = req.params.filename;

  // Basic security check to prevent directory traversal attacks
  if (filename.includes('..') || filename.includes('/') || filename.includes('\\')) {
    return res.status(400).send('Invalid filename');
  }

  const filePath = path.join(FILES_DIR, filename);

  // Check if file exists
  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      return res.status(404).send('File not found');
    }

    // Set headers to force download
    res.download(filePath, filename, (err) => {
      if (err) {
        res.status(500).send('Error downloading file');
      }
    });
  });
});

module.exports = router;